### PR TITLE
Implement Triangular, Honeycomb, and Kagome lattices

### LINKED ISSUE #31: Implement Triangular, Hexagonal, and Kagome Lattices

### PR DESCRIPTION
**Pull request by emerali**
Resolves #31

### CODE REVIEW

File: src/triangular.jl [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    end
+end
+
+
+for L in [:Honeycomb, :Kagome]
+
+    @eval struct $L{B <: NTuple{2, AbstractBoundary}} <: AbstractLattice
+        dims::NTuple{2, Int}
+        bcs::NTuple{2, AbstractBoundary}
+        translation_vectors::Dict{Tuple{Int, Int}, Vector{Coordinate{3, Int}}}
```
Comment (Roger-luo) [inline src/triangular.jl:17]: if we end up in using translation vectors to do it, should we actually have a type for this then wrap that type to be `Tri` , `Kagome` etc.?

Comment (emerali) [reply src/triangular.jl:17]: i dont understand, could you elaborate?

Comment (Roger-luo) [reply src/triangular.jl:17]: e.g should we have a container lattice type that looks like

```julia
struct TranslationLattice{N, B <: NTuple{N, AbstractBoundary}} <: AbstractLattice
   dims::NTuple{N, Int}
   bcs::B
   translation_vectors
end
```

?

File: src/triangular.jl [hunk 2]
```diff
@@ -0,0 +1,63 @@
+struct Triangular{B <: NTuple{2, AbstractBoundary}} <: AbstractLattice
+    dims::NTuple{2, Int}
+    bcs::NTuple{2, AbstractBoundary}
```
Comment (Roger-luo) [inline src/triangular.jl:3]:
```suggestion
    bcs::B
```

File: src/triangular.jl [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        new{B}(dims, bcs, Dict{Int, Vector{Coordinate{2, Int}}}())
+    end
+end
+
+
+for L in [:Honeycomb, :Kagome]
+
+    @eval struct $L{B <: NTuple{2, AbstractBoundary}} <: AbstractLattice
+        dims::NTuple{2, Int}
+        bcs::NTuple{2, AbstractBoundary}
```
Comment (Roger-luo) [inline src/triangular.jl:16]:
```suggestion
        bcs::B
```

**Comment by Roger-luo** [comment]
the other part looks good in general to me. But I need to learn about the translation vector technique from you first to have a better understanding on this.

**Pull request merged**